presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -155,22 +165,37 @@ public PlanNodeDecorrelator(PlanNodeIdAllocator idAllocator, Lookup lookup)
             }
 
             DecorrelationResult childDecorrelationResult = childDecorrelationResultOptional.get();
+
             if (childDecorrelationResult.atMostSingleRow) {
                 return childDecorrelationResultOptional;
             }
 
-            if (node.getCount() != 1) {
+            if (childDecorrelationResult.getConstantSymbols().isEmpty()) {
                 return Optional.empty();
             }
 
-            Set<Symbol> constantSymbols = childDecorrelationResult.getConstantSymbols();
+            if (node.getCount() > 1) {
```

Comment:
please invert the condition so methods declared below will in the same order as usage